diff --git a/cmake/macros.cmake b/cmake/macros.cmake
index d2067ab..de7306b 100644
--- a/cmake/macros.cmake
+++ b/cmake/macros.cmake
@@ -176,9 +176,13 @@ macro(SET_INSTALL_DIRS)
           set(LASZIP_LIB_INSTALL_DIR "lib")
       endif()
   endif ()
+  if (NOT DEFINED LASZIP_INCLUDE_INSTALL_ROOT)
     set(LASZIP_INCLUDE_INSTALL_ROOT "include/")
+  endif ()
+  if (NOT DEFINED LASZIP_INCLUDE_INSTALL_DIR)
     set(LASZIP_INCLUDE_INSTALL_DIR
         "${LASZIP_INCLUDE_INSTALL_ROOT}")
+  endif ()
     set(LASZIP_DOC_INCLUDE_DIR
         "share/doc/${PROJECT_NAME_LOWER}-${LASZIP_VERSION_MAJOR}.${LASZIP_VERSION_MINOR}")
     set(LASZIP_BIN_INSTALL_DIR "bin")
diff --git a/dll/CMakeLists.txt b/dll/CMakeLists.txt
index 4dd0ffa..83072c2 100644
--- a/dll/CMakeLists.txt
+++ b/dll/CMakeLists.txt
@@ -5,7 +5,7 @@ set(LASZIP_API_SOURCES
 
 LASZIP_ADD_LIBRARY(${LASZIP_API_LIB_NAME} ${LASZIP_API_SOURCES})
 
-install(DIRECTORY "${LASZIP_HEADERS_DIR}"
+install(DIRECTORY "${LASZIP_HEADERS_DIR}/"
         DESTINATION "${LASZIP_INCLUDE_INSTALL_DIR}"
         PATTERN "/CMakeFiles/*" EXCLUDE
 )
