diff --git a/cmake/macros.cmake b/cmake/macros.cmake
index d2067ab..de7306b 100644
--- a/cmake/macros.cmake
+++ b/cmake/macros.cmake
@@ -176,9 +176,13 @@ macro(SET_INSTALL_DIRS)
           set(LASZIP_LIB_INSTALL_DIR "lib")
       endif()
   endif ()
+  if (NOT DEFINED LASZIP_INCLUDE_INSTALL_ROOT)
     set(LASZIP_INCLUDE_INSTALL_ROOT "include/")
+  endif ()
+  if (NOT DEFINED LASZIP_INCLUDE_INSTALL_DIR)
     set(LASZIP_INCLUDE_INSTALL_DIR
         "${LASZIP_INCLUDE_INSTALL_ROOT}")
+  endif ()
     set(LASZIP_DOC_INCLUDE_DIR
         "share/doc/${PROJECT_NAME_LOWER}-${LASZIP_VERSION_MAJOR}.${LASZIP_VERSION_MINOR}")
     set(LASZIP_BIN_INSTALL_DIR "bin")
